src/interceptors/ClientRequest/utils/createRequest.ts

```diff
@@ -25,7 +25,7 @@ export function createRequest(clientRequest: NodeClientRequest): Request {
    * the request "Authorization" header.
    * @see https://github.com/mswjs/interceptors/issues/438
    */
-  if (clientRequest.url.username && clientRequest.url.password) {
+  if (clientRequest.url.username) {
     const auth = `${clientRequest.url.username}:${clientRequest.url.password}`
     headers.set('Authorization', `Basic ${btoa(auth)}`)
```

Comment:
Also, it's a good idea to add two more tests covering this:

1. username + no password set. Should result in the encoded `Basic USERNAME:` header.
2. user + password set explicitly to `""`. Should result in the same encoded header.

Reply:
I'll add them tomorrow

Reply:
Thank you so much!